### ISSUE TITLE
Roadmap towards TFLearn v0.3

### ISSUE THREAD

**Issue by aymericdamien** [tflearn/tflearn#111]
TFLearn v0.2 is going to be released soon (just need to finish writing some tutorials), so we can already think about features that would be great in the next TFLearn version:

- [ ] Switch layers 'name_scope' to 'op_scope' and add a custom argument ('var_scope') to share weights among different layers.
- [ ] Dynamic RNN support.
- [ ] TensorFlow QueueRunner support (So TFRecords can be used) (Moved from v0.2).

Any other ideas are welcome :)

**Comment by vinayakumarr** [comment]
probably you can also add Contextual LSTM (CLSTM) models for Large scale NLP tasks example

**Comment by daviddao** [comment]
@fdiehl and I are still eager to contribute the attention modules :)
Also, do you have any plans on incorporating RL into tflearn? 
When do you plan to release 0.3?

**Comment by TylerBalsam** [comment]
How about an implementation of RReLU? There's already one for PReLU, which is generally better than ReLU, but RReLU seems to improve even more upon them, notably by having less over fitting w/ small scale training sets. It shouldn't be incredibly difficult to implement.

(source: https://arxiv.org/pdf/1505.00853.pdf)

**Comment by aymericdamien** [comment]
@daviddao Yes! That would be nice :) Not so sure yet when 0.3 can be released, I guess when we nearly finish all milestones.
@TylerBalsam I checked the paper, it seems nice, I will see how we can implement it.

**Comment by wballard** [comment]
Shared weights is my vote.

**Comment by psnegi** [comment]
How about adding support for LMDB  and related utility for handling large data set with examples?
It will  speedup training for single machine usage. 
Creating hdf5 compressed data takes weeks  even on 8 core machine for large dataset like ilsvrc12.

**Comment by aymericdamien** [comment]
@wballard ok, will add it.
@psnegi With QueueRunner, it will be possible to use native TFRecords files and process raw data (such as images). LMDB format support would also be nice to have, will check this out, it shouldn't be hard to integrate it.

**Comment by aymericdamien** [comment]
@daviddao I integrated a first example of playing Atari games (Pacman by default) using OpenAI's gym environment. See: https://github.com/tflearn/tflearn/blob/master/examples/reinforcement_learning/atari_1step_qlearning.py.

**Comment by daviddao** [comment]
@aymericdamien Awesome! @fdiehl and I will try it out :)

**Comment by ppwwyyxx** [comment]
@aymericdamien Nice implementation, but seems not really related to tflearn. The only things it uses are the simple wrapper function for conv2d/fully_connect...

**Comment by aymericdamien** [comment]
@ppwwyyxx Yes, this is just a simple example with only the DQN and mean_square fn using tflearn. Later it would be nice to create a wrapper for RL tasks, same as what we did for generative models.

**Comment by EderSantana** [comment]
how about adversarial generative networks?
here is an example https://github.com/carpedm20/DCGAN-tensorflow
need to reuse weights to run the discriminator on real and sampled images

**Comment by aymericdamien** [comment]
@EderSantana That's a good idea, we can add it to the next version, it will be useful for many users.

**Comment by tobegit3hub** [comment]
Hi @aymericdamien , any update about `TFRecords` or `QueueRunner`?

**Comment by aymericdamien** [comment]
Hi @tobegit3hub, we plan to add new trainer functions that will support that, it will be release later, but we do not have an estimated date yet.

**Comment by onion-engineer** [comment]
I really look forward to GAN Example and Attention Module.
By the way, what about clipped relu? (min(max(z, 0), high))

I temporarily try it like

```python
def clipped_relu(z, high):
    output = z
    for _ in range(2):
        output = relu(output)
        output = - output + high
    return output
```

or

```python
def clipped_relu(z, low, high):
    output = z
    output = relu(output - low)
    output = relu(- output + (high - low))
    return - output + high
```

**Comment by TylerBalsam** [comment]
Any reason for the clipped relu? They seem to have fallen out of favor
recently due to their negative value gradient suppression.

Have you tried elu instead?

**Comment by ppwwyyxx** [comment]
@onion-engineer There is a `tf.clip_by_value` already.

**Comment by onion-engineer** [comment]
@TylerBalsam Yes, I use elu lately and it's awesome. I didn't think of using it instead of relu, but tanh because I wanted to see the difference in some image generation tasks.

@ppwwyyxx Thanks! I didn't know it existed.
